paddle/fluid/framework/data_layout_transform.cc

```diff
@@ -127,13 +127,17 @@ void TransDataLayoutFromMKLDNN(const OpKernelType& kernel_type_for_var,
       "TransDataLayoutFromMKLDNN only supports transform from MKLDNN to "
       "non-MKLDNN");
 
-  innerTransDataLayoutFromMKLDNN(in_layout, out_layout, in, out, place);
+#ifdef PADDLE_WITH_MKLDNN
+  innerTransDataLayoutFromMKLDNN(in_layout,
+                                 paddle::platform::get_cur_paddle_data_layout(),
+                                 in, out, place);
+#endif
 }
```

Comment:
Should the macro `#ifdef PADDLE_WITH_MKLDNN` be refined?
line 95-202 should be in one macro `#ifdef PADDLE_WITH_MKLDNN`, but there are three.

Reply:
I agree with @LeoZhao-Intel , I can try to reorganize code to be in PADDLE_WITH_MKLDNN , but in in  next part fo NHWC support PR (Conv & conv transpose NHWC support)